packages/drivers/routerlicious-driver/src/tokens.ts

```diff
@@ -3,26 +3,41 @@
  * Licensed under the MIT License.
  */
 
-import { ITokenClaims, ITokenProvider, ITokenService } from "@fluidframework/protocol-definitions";
-import jwtDecode from "jwt-decode";
+import { ITokenClaims } from "@fluidframework/protocol-definitions";
 
 /**
- * Extracts the claims contained within a token.
+ * The ITokenService abstracts the discovery of claims contained within a token
  */
-export class TokenService implements ITokenService {
-    public extractClaims(token: string): ITokenClaims {
-        return jwtDecode(token);
-    }
+export interface ITokenService {
+    extractClaims(token: string): ITokenClaims;
+}
+
+export interface ITokenResponse {
+    // JWT value
+    jwt: string;
+
+    // Flag indicating whether token was obtained from local cache
+    fromCache?: boolean;
 }
 
 /**
- * Checks the validation of a token.
+ * The ITokenProvider abstracts the token fetching mechanism for a host. Host will be responsible for
+ * implementing the interfaces.
  */
-export class TokenProvider implements ITokenProvider {
-    constructor(public token: string) {
-    }
+export interface ITokenProvider {
+    /**
+     * Fetches the orderer token from host
+     * @param refresh - Optional flag indicating whether token fetch must bypass local cache
+     * @returns TokenResponse object representing token value along with flag indicating
+     * whether token came from cache.
+     */
+    fetchOrdererToken(refresh?: boolean): Promise<ITokenResponse>;
 
-    public isValid(): boolean {
-        return (this.token !== undefined && this.token !== null);
-    }
+    /**
+     * Fetches the storage token from host
+     * @param refresh - Optional flag indicating whether token fetch must bypass local cache
+     * @returns TokenResponse object representing token value along with flag indicating
+     * whether token came from cache.
```

Comment:
Should we (in principle) support non-authenticated users (maybe not in FRS, but in general)?
You have some code that handles false returns from here, this is not in signature.

Reply:
This can be handled through tenantManager. tinylicious already does that today. So for something like tinylicious, we can return an empty jwt and service will still work.